packages/cozy-harvest-lib/src/helpers/konnectorBlock.js

```diff
@@ -0,0 +1,227 @@
+import { triggers } from 'cozy-client/dist/models/trigger'
+import { generateUniversalLink } from 'cozy-ui/transpiled/react/AppLinker'
+import get from 'lodash/get'
+
+import {
+  buildKonnectorQuery,
+  buildIsAccountConnectedQuery,
+  buildIsInErrorTriggersQuery,
+  buildTriggersByIdQuery
+} from './queries'
+import { hasNewVersionAvailable } from './konnectors'
+
+const formatKonnector = ({
+  client,
+  konnector,
+  universalLink,
+  iconStatus,
+  message
+}) => {
+  const link = generateUniversalLink({
+    cozyUrl: client.getStackClient().uri,
+    slug: universalLink.slug,
+    subDomainType: client.getInstanceOptions().cozySubdomainType,
+    nativePath: universalLink.nativePath
+  })
+
+  return {
+    name: konnector.name,
+    link,
+    vendorLink: konnector.vendor_link && {
+      component: 'a',
+      href: konnector.vendor_link,
+      target: '_blank'
+    },
+    iconStatus,
+    message
+  }
+}
+
+const getDataFromState = async (client, query) => {
+  try {
+    await client.query(query.definition, query.options)
+    const { data } = client.getQueryFromState(query.options.as)
+    return data[0]
+  } catch (error) {
+    throw error
+  }
+}
+
+const getKonnector = (client, query) => getDataFromState(client, query)
+
+const isAccountConnected = async (client, sourceAccount) => {
+  const query = buildIsAccountConnectedQuery(sourceAccount)
+  const data = await getDataFromState(client, query)
```

Comment:
```
const qDef =  buildIsAccountConnectedQuery(sourceAccount)
const { data } = useQuery(qDef.query, qDef)
return !!data
```

It seems like useQuery could improve a little bit its API to directly consume the options object that would contain the query. Then it would become a little bit easier to use:

```
```
const { data } = useQuery(buildIsAccountConnectedQuery(sourceAccount))
return !!data
```